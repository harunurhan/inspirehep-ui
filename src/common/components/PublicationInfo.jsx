import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';

class PublicationInfo extends Component {
  getPageOrArtidDisplay() {
    const { info } = this.props;
    if (info.has('page_start') && info.has('page_end')) {
      return `${info.get('page_start')}-${info.get('page_end')}`;
    } else if (info.has('page_start')) {
      return info.get('page_start');
    } else if (info.has('artid')) {
      return info.get('artid');
    }
    return null;
  }

  render() {
    const { info } = this.props;
    const material = info.get('material');
    const journalIssue = info.get('journal_issue');
    const pageOrArtidDisplay = this.getPageOrArtidDisplay();
    if (info.has('journal_title')) {
      return (
        <span>
          <i>{info.get('journal_title')}</i>
          {info.has('journal_volume') && (
            <span> {info.get('journal_volume')}</span>
          )}
          {info.has('year') && <span> ({info.get('year')})</span>}
          {journalIssue && <span> {journalIssue}</span>}
          {pageOrArtidDisplay && journalIssue && <span>,</span>}
          {pageOrArtidDisplay && <span> {pageOrArtidDisplay}</span>}
          {material && material !== 'publication' && <span> ({material})</span>}
        </span>
      );
    } else if (info.has('pubinfo_freetext')) {
      return <span>{info.get('pubinfo_freetext')}</span>;
    }
    return null;
  }
}

PublicationInfo.propTypes = {
  info: PropTypes.instanceOf(Map).isRequired,
};

export default PublicationInfo;
