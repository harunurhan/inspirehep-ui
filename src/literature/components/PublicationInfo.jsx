import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';

class PublicationInfo extends Component {
  render() {
    const { info } = this.props;
    if (info.has('journal_title')) {
      return (
        <span>
          <i>{info.get('journal_title')}</i>
          {info.has('journal_volume') && <span> {info.get('journal_volume')}</span>}
          {info.has('year') && <span> ({info.get('year')})</span>}
          {info.has('page_start') && info.get('page_end') && <span> {info.get('page_start')}-{info.get('page_end')}</span>}
          {info.has('page_start') && <span> {info.get('page_start')}</span>}
          {info.has('artid') && <span> {info.get('artid')}</span>}
        </span>
      );
    } else if (info.has('pubinfo_freetext')) {
      return (
        <span>{info.get('pubinfo_freetext')}</span>
      );
    }
    return null;
  }
}

PublicationInfo.propTypes = {
  info: PropTypes.instanceOf(Map).isRequired,
};

export default PublicationInfo;
