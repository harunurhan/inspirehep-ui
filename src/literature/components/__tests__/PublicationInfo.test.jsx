import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import PublicationInfo from '../PublicationInfo';

describe('PublicationInfo', () => {
  it('renders with journal_title present', () => {
    const info = fromJS({
      journal_title: 'Test Journal',
    });
    const wrapper = shallow((
      <PublicationInfo info={info} />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with journal_title and alll others fields', () => {
    const info = fromJS({
      journal_title: 'Test Journal',
      journal_volume: 'TV',
      year: 2016,
      page_start: '1',
      page_end: '2',
      artid: '012345',
    });
    const wrapper = shallow((
      <PublicationInfo info={info} />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with journal_title and alll others fields', () => {
    const info = fromJS({
      journal_title: 'Test Journal',
      journal_volume: 'TV',
      year: 2016,
      page_start: '1',
      page_end: '2',
      artid: '012345',
      pubinfo_freetext: 'Test. Pub. Info. Freetext',
    });
    const wrapper = shallow((
      <PublicationInfo info={info} />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with pubinfo_freetext', () => {
    const info = fromJS({
      pubinfo_freetext: 'Test. Pub. Info. Freetext',
    });
    const wrapper = shallow((
      <PublicationInfo info={info} />
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without pubinfo_freetext or journal_title', () => {
    const info = fromJS({});
    const wrapper = shallow((
      <PublicationInfo info={info} />
    ));
    expect(wrapper).toMatchSnapshot();
  });
});
