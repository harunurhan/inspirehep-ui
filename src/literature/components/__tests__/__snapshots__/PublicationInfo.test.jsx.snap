// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicationInfo renders with journal_title and alll others fields 1`] = `
<span>
  <i>
    Test Journal
  </i>
  <span>
     
    TV
  </span>
  <span>
     (
    2016
    )
  </span>
  <span>
     
    1
    -
    2
  </span>
  <span>
     
    1
  </span>
  <span>
     
    012345
  </span>
</span>
`;

exports[`PublicationInfo renders with journal_title and alll others fields 2`] = `
<span>
  <i>
    Test Journal
  </i>
  <span>
     
    TV
  </span>
  <span>
     (
    2016
    )
  </span>
  <span>
     
    1
    -
    2
  </span>
  <span>
     
    1
  </span>
  <span>
     
    012345
  </span>
</span>
`;

exports[`PublicationInfo renders with journal_title present 1`] = `
<span>
  <i>
    Test Journal
  </i>
</span>
`;

exports[`PublicationInfo renders with pubinfo_freetext 1`] = `
<span>
  Test. Pub. Info. Freetext
</span>
`;

exports[`PublicationInfo renders without pubinfo_freetext or journal_title 1`] = `""`;
